geopandas/tests/test_geodataframe.py

```diff
@@ -607,6 +607,20 @@ def test_geodataframe_geointerface(self):
         assert self.df.__geo_interface__['type'] == 'FeatureCollection'
         assert len(self.df.__geo_interface__['features']) == self.df.shape[0]
 
+    def test_geodataframe_iterfeatures(self):
+        df = self.df.iloc[:1].copy()
+        df.ix[0, 'BoroName'] = np.nan
+        # when containing missing values
+        # null: ouput the missing entries as JSON null
+        result = list(df.iterfeatures(na='null'))[0]['properties']
+        assert result['BoroName'] is None
+        # drop: remove the property from the feature.
+        result = list(df.iterfeatures(na='drop'))[0]['properties']
+        assert 'BoroName' not in result.keys()
+        # keep: output the missing entries as NaN
+        result = list(df.iterfeatures(na='keep'))[0]['properties']
+        assert np.isnan(result['BoroName'])
```

Comment:
Can you add one more specific test for checking that the other (non-null) features are python scalars and not numpy scalars? 

Like 

```
n [56]: df = geopandas.read_file(geopandas.datasets.get_path('nybb'))                                                                                                          

In [58]: type(df.loc[0, 'Shape_Leng'])                                                                                                                                          
Out[58]: numpy.float64

In [59]: type(df.loc[0, 'Shape_Leng']) is float                                                                                                                                 
Out[59]: False
```

The above should be true when accessing one of the elements in the features dict. 

And can you do this also when only having numerical columns? eg with `df[['Shape_Leng', 'Shape_Area', 'geometry']].iterfeatures()`

Reply:
Thank you for the review.
I pushed the changes following your comments.

Reply:
Thanks for the update!